doc/source/whatsnew/v2.0.0.rst

```diff
@@ -291,6 +291,37 @@ The new behavior, as for datetime64, either gives exactly the requested dtype or
    ser.astype("timedelta64[s]")
    ser.astype("timedelta64[D]")
 
+.. _whatsnew_200.api_breaking.zero_len_indexes:
+
+Empty DataFrames/Series will now default to have a ``RangeIndex``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When users instantiate :class:`DataFrame` or :class:`Series` objects without specifying the axes (``index=None``
+and/or ``columns=None``), pandas infers the index type for them. Previously, when instantiating
```

Comment:
Is it critical to describe the inference of the index type here? IMO it appears the more important context here is when the constructed DataFrame or Series has length 0?

Reply:
Yeah, I'm a bit in doubt how to describe it. I'll try basing it on your text.

The change in this PR is (with index=None and/or columns=None):
1. if the data is dictlike (dicts, series etc) the index/columns will be inferred from the dict keys (unchanged functionality compared to before this PR)
2. if the data is listlike or None the index/columns will be inferred to `RangeIndex` (before `RangeIndex` was used in the the non-zero length case, while `Index` with object dtype was inferred in the zero-length case ).

This PR only deals with the second case and ensures a `RangeIndex` is always used there.

Reply:
Got it, this explanation greatly helped me understand whats changing.

> if the data is listlike or None the index/columns will be inferred to RangeIndex

I think this would be key to include in the whatsnew entry